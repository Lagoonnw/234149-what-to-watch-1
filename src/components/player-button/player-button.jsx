import React from 'react';
import PropTypes from 'prop-types';

export const PlayerButton = ({isPlaying = false, onClick}) => {
  return (
    <button type="button" className="player__play" onClick={onClick}>
      {(isPlaying) ?
        <svg viewBox="0 0 14 21" width="14" height="21"><use xlinkHref="#pause"/></svg> :
        <svg viewBox="0 0 19 19" width="19" height="19"><use xlinkHref="#play-s"/></svg>
      }
      <span>{(isPlaying) ? `Pause` : `Play`}</span>
    </button>
  );
};

PlayerButton.propTypes = {
  isPlaying: PropTypes.bool,
  onClick: PropTypes.func.isRequired
};
