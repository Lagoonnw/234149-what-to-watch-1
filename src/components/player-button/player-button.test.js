import React from 'react';
import renderer from 'react-test-renderer';
import {PlayerButton} from './player-button.jsx';

test(`Should render Player button correctly`, () => {
  const tree = renderer.create(
      <PlayerButton onClick={jest.fn()}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
