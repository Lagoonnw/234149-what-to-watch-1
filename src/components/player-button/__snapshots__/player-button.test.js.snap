// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Player button correctly 1`] = `
<button
  className="player__play"
  onClick={[MockFunction]}
  type="button"
>
  <svg
    height="19"
    viewBox="0 0 19 19"
    width="19"
  >
    <use
      xlinkHref="#play-s"
    />
  </svg>
  <span>
    Play
  </span>
</button>
`;
